buildengine/engine.go

```diff
@@ -29,6 +29,10 @@ type schemaChange struct {
 	*schema.Module
 }
 
+type Listener struct {
```

Comment:
This should be an interface, it's very rare that you would need to use a struct for something like this.